### PR TITLE
Added Symfony 4 compatibility and refactored code

### PR DESCRIPTION
**Pull request by junaidbinfarooq**
Replaced Symfony\Bundle\FrameworkBundle\Controller\Controller with Symfony\Bundle\FrameworkBundle\Controller\AbstractController
Added proper return type and type hints
Updated php doc-blocks

### CODE REVIEW

**Comment by XWB** [comment]
You will need to replace the `$this->get` calls as well, because `AbstractController` uses a smaller container that does not contain all services.

File: Controller/EditInPlaceController.php [hunk 1]
```diff
@@ -22,15 +22,16 @@
 /**
  * @author Damien Alexandre <dalexandre@jolicode.com>
  */
-class EditInPlaceController extends Controller
+class EditInPlaceController extends AbstractController
 {
     /**
+     * @param Request $request
      * @param string $configName
      * @param string $locale
```
Comment (welcoMattic) [inline Controller/EditInPlaceController.php]: cs-fixer should remove those PHPDoc lines since they are redundant with type hinting.

Comment (rvanlaak) [reply Controller/EditInPlaceController.php]: #342 also has the commits that fix cs

Comment (welcoMattic) [reply Controller/EditInPlaceController.php]: Ok, let's merge #342 and this one. :+1:

**Comment by junaidbinfarooq** [comment]
> Thanks for your first contribution. :) It looks great. 
> Could you please update CS?

Updated the PR with CS-fixes and more changes. Could you please have a look.

**Comment by XWB** [comment]
@junaidbinfarooq Much better indeed. Still, those changes will only work if you have autowire enabled. So you'll have to create `Resources/config/controller.yml` and add each controller as a service.

**Comment by junaidbinfarooq** [comment]
> @junaidbinfarooq Much better indeed. Still, those changes will only work if you have autowire enabled. So you'll have to create `Resources/config/controller.yml` and add each controller as a service.

Oh! yeah. Done now. Please have a look.

File: Resources/config/services.yml [hunk 2]
```diff
@@ -1,3 +1,6 @@
+imports:
+    - { resource: controller.yml }
```
Comment (XWB) [inline Resources/config/services.yml]: Being a bundle, it would be better to move the import to `TranslationExtension`. See https://github.com/php-translation/symfony-bundle/blob/master/DependencyInjection/TranslationExtension.php#L42

File: Resources/config/controller.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+services:
+    _defaults:
```
Comment (XWB) [inline Resources/config/controller.yml]: The `_defaults` section needs to be removed. Autowire must be enabled in your own application and not in a bundle.

File: composer.json [hunk 4]
```diff
@@ -22,7 +22,8 @@
         "php-translation/symfony-storage": "^1.0",
         "php-translation/extractor": "^1.6",
         "nyholm/nsa": "^1.1",
-        "twig/twig": "^1.42 || ^2.11"
+        "twig/twig": "^1.42 || ^2.11",
+        "ext-json": "*"
```
Comment (XWB) [inline composer.json]: Is this change required?

**Comment by XWB** [comment]
@junaidbinfarooq Almost done :)

File: Resources/config/controller.yml [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+        autowire: true
+        autoconfigure: true
+        public: true
+
+    Translation\Bundle\Controller\:
```
Comment (XWB) [inline Resources/config/controller.yml]: You will need to inject all the services into the constructor, so the controllers can work without autowire.

File: composer.json (hunk 4 above)
Comment (junaidbinfarooq) [reply composer.json]: Added this after encountering few usages of json_encode/decode functions. PhpStorm does show a warning for this.

File: Resources/config/services.yml (hunk 2 above)
Comment (junaidbinfarooq) [reply Resources/config/services.yml]: Did see this before but still thought of importing the services as we do in the applications. Anyways, this is done now.

File: Resources/config/controller.yml (hunk 3 above)
Comment (junaidbinfarooq) [reply Resources/config/controller.yml]: Done.

File: Resources/config/controller.yml (hunk 5 above)
Comment (junaidbinfarooq) [reply Resources/config/controller.yml]: Have indeed used constructor injection.

**Comment by welcoMattic** [comment]
@junaidbinfarooq Hi, as https://github.com/php-translation/symfony-bundle/pull/342 is now merged in master, could you fix conflicts due to the merge (https://github.com/php-translation/symfony-bundle/pull/342#issuecomment-559004353). 
Thank you :+1:

**Comment by junaidbinfarooq** [comment]
> @junaidbinfarooq Hi, as #342 is now merged in master, could you fix conflicts due to the merge ([#342 (comment)](https://github.com/php-translation/symfony-bundle/pull/342#issuecomment-559004353)).
> Thank you

Done. Could you please have a look.

**Comment by XWB** [comment]
Looks good in general. Just some tests need to be fixed.

**Comment by junaidbinfarooq** [comment]
@odolbeau Couldn't you create a fork of my branch and update the tests instead of creating another PR and closing this one?

**Comment by odolbeau** [comment]
[I tried](https://github.com/php-translation/symfony-bundle/pull/346#issuecomment-559785211) but I finally choose to create a new branch for several reasons.
Your branch mix 2 things: the Symfony 5 migration & the Strict type hinting. This branch was quite outdated compared to master when I started to work. Last but not least, you weren't active since a few weeks.
I hope you'll understand why I did that.
Thanks for your work anyway! :)
